### PR TITLE
Assorted Notary mark ups

### PR DESCRIPTION
**Pull request by darracott**
Implementing hangover review comments from https://github.com/microsoft/scitt-ccf-ledger/pull/73 addressing https://github.com/microsoft/scitt-ccf-ledger/issues/84

Featuring:
- x5chain (de)serialization 
- QCBOR tweaks
- Cleaning up redundant code  
- Notary spec allows x5chain in the protected header (but currently notation always puts it in the unprotected headers). We now check for the x5chain in the protected header of notary claims before the unprotected header.

### CODE REVIEW

File: app/src/verifier.h [hunk 1]
```diff
@@ -273,22 +268,35 @@ namespace scitt::verifier
       // Validate protected header
       validate_notary_protected_header(phdr, configuration);
 
-      // Validate unprotected header
-      // Unprotected header must contain x5chain else the notary claim cannot be
-      // verified.
-      if (!uhdr.x5chain.has_value())
+      std::vector<std::vector<uint8_t>> x5chain{};
+
+      if (phdr.x5chain.has_value() and uhdr.x5chain.has_value())
```
Comment (letmaik) [inline app/src/verifier.h:273]: Let's use `&&` for consistency.

File: app/src/cose.h [hunk 2]
```diff
@@ -430,6 +421,12 @@ namespace scitt::cose
     auto phdr = decode_protected_header(ctx);
     auto uhdr = decode_unprotected_header(ctx);
 
+    QCBORDecode_ExitArray(&ctx);
```
[the remark this answers is not in the document]
Comment (darracott) [reply app/src/cose.h:424]: https://github.com/laurencelundblade/QCBOR/blob/37286c08a00d7516cf2da173a48d1bfdd5260eff/inc/qcbor/qcbor_spiffy_decode.h#L635 it's okay to exit the array without consuming the final two items

[the remark this answers is not in the document]
Comment (letmaik) [reply app/src/cose.h:424]: Got it, wasn't aware that this is part of the "spiffy" API.

File: app/src/cose.h [hunk 3]
```diff
@@ -430,6 +422,12 @@ namespace scitt::cose
...
+    auto error = QCBORDecode_Finish(&ctx);
+    if (error)
+    {
+      throw std::runtime_error("Failed to decode COSE_Sign1");
```
Comment (plietar) [inline app/src/cose.h:429]:
```suggestion
      throw COSEDecodeError("Failed to decode COSE_Sign1");
```

**Review by plietar — APPROVED** [review]

**Pull request merged**